### DISCUSSION TITLE
Acknowledge publisher that message is received by subscriber.

### DISCUSSION THREAD

**Discussion by moxit0** [nats-io/nats.java#788] [Q&A]
Hi folks,

We are impressed by NATS and decided to replace our messaging infrastructure by it for some of our services but we run into problem. I am trying to achieve request/reply approach by jetstream and reading documentation I found: [exactly-once-delivery](https://docs.nats.io/using-nats/developer/develop_jetstream/model_deep_dive#exactly-once-delivery) and see diferent Acknowledgemnt models. So is it possible publisher to receive acknowledge by subscriber with double acks ? If is not possible does my only option is to respond to the publisher on separate subject ? To be honest is quite  complex for beginner to prepare push consumer / push subscriber or pull consumer pull subscriber so would be very convenient if NATS Java Client have separate Interface/Class/Builder whatever that makes  these easier.

**Comment by scottf** [comment]
Yes, I think your best option, for confirmation of a message being processed, is to publish a message back to the original publisher. There are some examples here https://github.com/nats-io/java-nats-examples. The chain-of-command and core-reply-request-patterns maybe be the most relevant.

**Reply by moxit0** [reply]
Thank you, will use some output subject for message response.
